### PR TITLE
module 2

### PR DESCRIPTION
**Pull request by LaiqianDS**
Habría que orientar más el EDA a contrastar hipótesis

### CODE REVIEW

File: src/module_2/part1.md [hunk 1]
```diff
@@ -0,0 +1,1380 @@
...
+# EDA
+
+## Abandoned Carts Analysis
+
+We are going to check the size of abandoned carts
+
+
+```python
+# Calculate abandoned carts size and its frecuency
+abandoned_cart["cart_size"] = abandoned_cart["variant_id"].apply(lambda x: len(x))
```
Comment (fsanzestevez) [inline src/module_2/part1.md:559]: We could directly do:
`abandoned_cart["cart_size"] = abandoned_cart["variant_id"].apply(len)`

File: src/module_2/part1.md [hunk 2]
```diff
@@ -0,0 +1,1380 @@
...
+abandoned_cart_size_frecuency = abandoned_cart["cart_size"].value_counts()
+
+plt.style.use(style="seaborn-v0_8-darkgrid")
+abandoned_cart_size_frecuency.sort_index().plot(
+    kind="bar", figsize=(10, 5), title="Abandoned Cart Size"
+)
+pass
```
Comment (fsanzestevez) [inline src/module_2/part1.md:566]: Why this `pass` ?

File: src/module_2/part1.md [hunk 3]
```diff
@@ -0,0 +1,1380 @@
...
+
+We are going to check the size of abandoned carts
+
+
+```python
+# Calculate abandoned carts size and its frecuency
+abandoned_cart["cart_size"] = abandoned_cart["variant_id"].apply(lambda x: len(x))
+abandoned_cart_size_frecuency = abandoned_cart["cart_size"].value_counts()
+
+plt.style.use(style="seaborn-v0_8-darkgrid")
```
Comment (fsanzestevez) [inline src/module_2/part1.md:562]: If this is something we want to use throughout the whole notebook, maybe we should move it to the top (after imports, and with the CONSTANTs initialisations.

File: src/module_2/part1.md [hunk 4]
```diff
@@ -0,0 +1,1380 @@
...
+
+## Abandoned Carts Analysis
+
+We are going to check the size of abandoned carts
+
+
+```python
+# Calculate abandoned carts size and its frecuency
+abandoned_cart["cart_size"] = abandoned_cart["variant_id"].apply(lambda x: len(x))
+abandoned_cart_size_frecuency = abandoned_cart["cart_size"].value_counts()
```
Comment (fsanzestevez) [inline src/module_2/part1.md:560]: frequency, with 'q'

File: src/module_2/part1.md (hunk 2 above)
Comment (fsanzestevez) [reply src/module_2/part1.md:566]: You may want to add `plt.show()` after each plot to ensure the plots are displayed correctly. Maybe this is why you added `pass` ?

File: src/module_2/part1.md [hunk 5]
```diff
@@ -0,0 +1,1380 @@
...
+
+
+    
+![png](part1_files/part1_35_0.png)
+    
+
+
+
+```python
+products_ordered["product_type"].value_counts()[:10].plot(
```
Comment (fsanzestevez) [inline src/module_2/part1.md:995]: This is repeated from `Line 943`, no?

File: src/module_2/part2.md [hunk 6]
```diff
@@ -0,0 +1,710 @@
...
+     24  days_since_purchase_product_type  float64
+     25  avg_days_to_buy_product_type      float64
+     26  std_days_to_buy_product_type      float64
+    dtypes: float64(19), int64(4), object(4)
+    memory usage: 593.4+ MB
+
+
+
+```python
+# Check for missing values
```
Comment (fsanzestevez) [inline src/module_2/part2.md:234]: Unnecessary comment

File: src/module_2/part2.md [hunk 7]
```diff
@@ -0,0 +1,710 @@
...
+    1.0      10456
+    Name: count, dtype: int64
+    
+
+
+Let's see if these variables have correlation
+
+
+```python
+# Heatmap of binary columns
```
Comment (fsanzestevez) [inline src/module_2/part2.md:569]: Unnecessary comment

File: src/module_2/part2.md [hunk 8]
```diff
@@ -0,0 +1,710 @@
...
+
+```python
+count_columns = ["count_adults", "count_children", "count_pets", "count_babies"]
+```
+
+
+```python
+# Drop repeated users, we have to bear in mind that this is not the best way to handle repeated users
+# However, for the purpose of this example, we will do it
+# The objective is to describe "count_columns" per user
```
Comment (fsanzestevez) [inline src/module_2/part2.md:593]: Good! But maybe a docstring of markdown if we are writing long comments

**Review by fsanzestevez — APPROVED** [review]
Overall good!
Just keep in mind comments should add value.
And the `pass`'s after the plots.

File: src/module_2/part1.md (hunk 2 above)
Comment (LaiqianDS) [reply src/module_2/part1.md:566]: The intention of the `pass` was just to avoid the output of the cell, which might be annoying

**Pull request merged**